oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/NodeImpl.java

```diff
@@ -1679,6 +1679,24 @@ public PropertyImpl performNullable() throws RepositoryException {
         });
     }
 
+    @Override
+    public Node getParentOrNull() throws RepositoryException {
+        return perform(new NodeOperation<Node>(dlg, "getParentOrNull") {
+            @Override
+            public Node perform() throws RepositoryException {
```

Comment:
i would also recommend to annotated the return value with @NotNull.
the method should be renamed to `performNullable` for consistency

Reply:
Done